src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -218,16 +225,23 @@ class MarathonSchedulerService @Inject() (
       }
     } onComplete { result =>
       synchronized {
-        driver = None
 
         log.info(s"Driver future completed with result=$result.")
         result match {
           case Failure(t) => log.error("Exception while running driver", t)
           case _ =>
         }
 
-        // tell leader election that we step back, but want to be re-elected if isRunning is true.
-        electionService.abdicateLeadership(error = result.isFailure, reoffer = isRunningLatch.getCount > 0)
+        // ONLY do this if there's some sort of driver crash: avoid invoking abdication logic if
+        // the driver was stopped via stopDriver. stopDriver only happens when
+        //   1. we're being terminated (and have already abdicated)
+        //   2. we've lost leadership (no need to abdicate if we've already lost)
+        driver.foreach { _ =>
```

Comment:
So in the case we get elected, and fail during driver creation, we would never abdicate?